prov/efa/src/efa_ep.c

```diff
@@ -228,14 +228,20 @@ static struct efa_ep *efa_ep_alloc(struct fi_info *info)
 
 static void efa_ep_destroy(struct efa_ep *ep)
 {
+	int err;
+
 	if (ep->self_ah)
 		ibv_destroy_ah(ep->self_ah);
 
 	efa_ep_destroy_qp(ep->qp);
 	fi_freeinfo(ep->info);
 	free(ep->src_addr);
-	if (ofi_endpoint_close(&ep->util_ep))
-		FI_WARN(&efa_prov, FI_LOG_EP_CTRL, "Unable to close util EP\n");
+	if (ep->util_ep.domain) {
```

Comment:
It feels odd to check for the value of the domain to decide if we need to call `ofi_endpoint_close`. 
The base error that we are addressing here seems to occur since we call `efa_ep_destroy` when `ofi_endpoint_init` fails in `efa_ep_open`. It seems we should be able to track ourselves if that call was successful (and thus if we need to call the ofi-level endpoint close).

Reply:
Make sense. I will add a flag in `efa_ep` to track `ofi_endpoint_init` status.